contrib/endpoints/src/api_manager/config.cc

```diff
@@ -113,6 +113,36 @@ MethodInfoImpl *Config::GetOrCreateMethodInfoImpl(const string &name,
   return i->second.get();
 }
 
+bool Config::LoadQuotaRule(ApiManagerEnvInterface *env) {
+  for (const auto &rule : service_.quota().metric_rules()) {
+    auto method = utils::FindOrNull(method_map_, rule.selector());
+    if (method) {
+      for (auto &metric_cost : rule.metric_costs()) {
+        (*method)->add_metric_cost(metric_cost.first, metric_cost.second);
+      }
+    } else {
+      env->LogDebug("Method not Found.");
+    }
+  }
+
+  for (const auto &rule : service_.quota().rules()) {
```

Comment:
There is no grouprules field in quota: https://github.com/cloudendpoints/service-control-client-cxx/blob/rate_limiting/proto/google/api/quota.proto